### PR TITLE
12752 create categories screen

### PR DESCRIPTION
**Pull request by tuandao2511**
## [Pull request title]
Create category screen
### [Redmine link]
https://edu-redmine.sun-asterisk.vn/issues/12752

### CODE REVIEW

File: app/src/main/java/com/asterisk/tuandao/themoviedb/ui/genre/GenreAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        genres = newGenres
+        notifyDataSetChanged()
+    }
+
+    class GenreViewHolder(val binding: ItemGenreMovieBinding) : RecyclerView.ViewHolder(binding.root) {
+
+        fun bindView(data: Genre?, genreViewModel: GenreViewModel) {
+            binding.run {
+                this.genre = data
+                this.genreViewModel = genreViewModel
```
Comment (hieubnv-0865) [inline app/src/main/java/com/asterisk/tuandao/themoviedb/ui/genre/GenreAdapter.kt]: a nên tạo genreViewModel trong constructor của GenreViewHolder nhé

File: app/src/main/res/layout/item_genre_movie.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        xmlns:card_view="http://schemas.android.com/apk/res-auto">
+    <data>
+        <variable name="genre"
+                  type="com.asterisk.tuandao.themoviedb.data.source.model.Genre"/>
+        <variable name="genreViewModel" type="com.asterisk.tuandao.themoviedb.ui.genre.GenreViewModel"/>
+    </data>
+    <androidx.cardview.widget.CardView
+            android:id="@+id/cardGenre"
+            android:layout_width="match_parent"
+            android:layout_height="50dp"
```
Comment (hieubnv-0865) [inline app/src/main/res/layout/item_genre_movie.xml]: hard code

File: app/src/main/res/layout/fragment_genre.xml [hunk 3]
```diff
@@ -1,7 +1,36 @@
...
+                android:layout_height="wrap_content">
+            <com.google.android.material.appbar.CollapsingToolbarLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="match_parent"
+                    app:contentScrim="@color/color_primary"
+                    app:layout_scrollFlags="scroll|exitUntilCollapsed">
+                    <ImageView
+                            android:id="@+id/imageGenreBackground"
+                            android:layout_width="match_parent"
+                            android:layout_height="300dp"
```
Comment (hieubnv-0865) [inline app/src/main/res/layout/fragment_genre.xml]: hard code

File: app/src/main/java/com/asterisk/tuandao/themoviedb/ui/home/HomeFragment.kt [hunk 4]
```diff
@@ -59,7 +57,8 @@ class HomeFragment : BaseFragment(), HomeMovieNavigator {
     }
 
     private fun doObserve() {
-        mainViewModel.movie?.observe(this as Fragment, Observer {
+//        viewDataBinding.viewmodel?.getMovies()
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/asterisk/tuandao/themoviedb/ui/home/HomeFragment.kt]: remove comment or comment meaningfully

File: app/src/main/java/com/asterisk/tuandao/themoviedb/ui/main/MainActivity.kt [hunk 5]
```diff
@@ -14,6 +15,7 @@ import kotlinx.android.synthetic.main.activity_main.*
 import javax.inject.Inject
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/asterisk/tuandao/themoviedb/ui/main/MainActivity.kt]: remove 2 bl

**Pull request merged**